testing/__init__.py

```diff
@@ -48,13 +48,17 @@ def print_signals(args=()):
 
 def child_pids(pid):
     """Return a list of direct child PIDs for the given PID."""
-    pid = str(pid)
-    tasks = LocalPath('/proc').join(pid, 'task').listdir()
-    return set(
-        int(child_pid)
-        for task in tasks
-        for child_pid in task.join('children').read().split()
-    )
+    children = set()
+    for p in LocalPath('/proc').listdir():
+        stat = p.join('stat')
+        if stat.isfile():
+            stat = stat.open().read()
+            m = re.match('^\d+ \([^\)]+\) [a-zA-Z] (\d+) ', stat)
+            assert m, stat
+            ppid = int(m.group(1))
+            if ppid == pid:
+                children.add(int(p.basename))
+    return children
```

Comment:
This looks worse, but turns out the `children` file is pretty scary:

```
<TheJH> ckuehl: heh, the source code for the /children file says it skips processes when dealing with a live process whose children exit while you're reading the file
<TheJH> ckuehl: because it seeks back to its last position by just skipping over N children :D
<TheJH> ckuehl: because it was written for CRIU and they never needed any stronger guarantees :D
<TheJH> ckuehl: so, uh, don't use that? :D
```

(thanks again @thejh!)

We don't need strong guarantees against races here since it's just a test (both the old and new code are racy). Plus, this works on non-Linux and is how `pgrep -P` does it.